Dragging the Globe

CesiumBen opened the pull request:
<!--
    * Please fill out every section in this template.
    * If this is a draft PR for early review, answer as much as possible and revisit the template when ready for full review.
-->

### Summary

This PR adds the dragging the globe functionality. 

<!--
In your own words, please summarize the pull request, what changes were made and why, both at end-user and technical levels.
Provide as much context as needed for the reviewer to understand the pull request. Are there specific files that should get
special attention?
-->

### Author checklist

- Tutorial for #5 <!-- After submitting, verify that the issue was linked to this PR. -->
- Has the new tutorial been integrated into the main level? Yes
- Has a new section of the readme been added for this tutorial? Yes
- Have you done a full self-review of the code and blueprints? Yes
- Have you tagged a primary reviewer? Yes
- Are there any remaining tasks to do or blocking questions to answer before we can merge this? No
<!--
  If "yes", create a task list:
  - [ ] Task 1
  - [ ] Blocking Question 1
-->

### Testing plan

Open up the dragging the globe level and grab the earth with either grip trigger. Test moving the head around as well. 

In the main level you can return to normal view by teleporting on the Earth by pointing to teleport as you normally would on the ground. While in the main level you can also enter the "Earth View" mode by flying high enough. 

<!--
  * If this PR fixes a bug, list the exact steps to reproduce the original issue.
     * Include links to any required data or screenshots.
     * Include any edge cases such as user error, invalid data, etc...
  * Identify performance implications of changes and how to test (if any).
If manual testing is required:
  * List testing steps that are particular to this code change.
-->

<!--
  Everything below is a check-list for the reviewer. Leave it here for them.
-->

### Reminders for reviewers

Thank you for taking the time to review this PR. By approving a PR you are taking as much responsibility for these changes as the author. Please keep these points in mind throughout the review process:

- Review and run all parts of the test plan on this branch and verify it matches expectations.
- Review the code and make sure you do not have any remaining questions or concerns. You should understand why this code change fixes the issue and agree that it's the best approach to doing so.
- Review the automated tests and make sure there are no missing tests or edge cases.
- When you feel this is ready bump it to a second reviewer or merge.

xuelongmu commented:
Thanks Ben! Sorry for the delayed review on this. I noticed a few things while testing in standalone:

1) Loading the terrain took a long time, about one minute or so before the continents appeared and I wasn't just looking at a blue ball. Might be worth making a note of so users don't get confused (subsequent loads were much faster, indicating this was a matter of caching downloaded tiles)
2) Unfortunately, having an object fixed to the center of your vision felt nauseating in VR. I think it'd be less nauseating if it was in front of you but persisted in the same world-space location, so it's acting more like an object in the real world.
3) Picking up the globe with your controller was also a little too jittery for a comfortable experience, and I started feeling nauseous there too. I think we maybe we could have "dragging" simply rotate the globe in-place (similar to spinning the globe by swiping your phone touchscreen in CesiumJS) instead of treating the globe as something you can pick up and move around.

CesiumBen commented:
Thank you @xuelongmu! 

Great points. The dragging functionality has been removed for now and will be added in after some fine tuning so that it runs smoothly. The Earth viewer now positions the earth in front of the player while transitioning, but after that you can move freely around the globe or rotate it with thumbstick input. 

Ready for re-review.

xuelongmu commented:
Thanks @CesiumBen! This looks great overall. Few minor things:

![image](https://user-images.githubusercontent.com/10490807/191835048-a9f28768-ba5b-45ff-9f45-3409196c5a9d.png)

On the `BP_VRPawn_EarthViewer`, should `SetEarthViewCoordinates` automatically call `PositionEarthInView`? This would be a cleaner API as the developer wouldn't need to worry about calling the second function every time. Unless there's a situation where you're setting coordinates but delaying updating the view.

![image](https://user-images.githubusercontent.com/10490807/191835429-f721240b-346d-40f3-9a11-c01a2bf4a6ea.png)

Similar feedback, it seems like the EarthViewer component should be able to update its own position on Tick rather than have the developer remember to put it on its owning actor's Tick. This comment in the SetEarthView function would also make a bit more sense:

![image](https://user-images.githubusercontent.com/10490807/191835652-7da852ce-5276-451f-92f5-16f8105b6741.png)

as after reading that comment, I was looking for a Tick function on the component but that functionality is handled on the tick of the owning actor.

Finally, quick naming thing - BP_EarthViewer should have Component appended to the end.

CesiumBen commented:
Thanks @xuelongmu, updates below

_____

> BP_EarthViewer should have Component appended to the end.

Updated, thanks for catching that

![BP_EarthViewerComponent](https://user-images.githubusercontent.com/79323980/191838257-d6478092-a827-43df-8be4-2eb3a21b0441.png)

_____


> On the BP_VRPawn_EarthViewer, should SetEarthViewCoordinates automatically call PositionEarthInView? This would be a cleaner API as the developer wouldn't need to worry about calling the second function every time. Unless there's a situation where you're setting coordinates but delaying updating the view.

I've updated the "SetEarthViewCoordinates" function to also set the Earth in view, moving that functionality from the input graph. I also added in a check to only set the Earth in view if you are in Earth view, so that if you're in normal view you can update safely update your coordinates without any Earth visualization. 

![UpdatedInputOnPawn](https://user-images.githubusercontent.com/79323980/191839721-29f05c4a-2ad0-411a-af37-a69bfacdab73.png)

![CoordinatesSetEarthInView](https://user-images.githubusercontent.com/79323980/191839708-b172b1fb-cca6-423b-ba16-a73e220bc291.png)


_____

>Similar feedback, it seems like the EarthViewer component should be able to update its own position on Tick rather than have the developer remember to put it on its owning actor's Tick. This comment in the SetEarthView function would also make a bit more sense:

I've moved it to the tick of its own component, earlier control of that tick was needed because of the grabbing which is no longer applied. In the future we'd want to disable tick of the component dynamically.

![EarthViewerTick](https://user-images.githubusercontent.com/79323980/191839868-ceb80701-790b-4522-9dbc-dbb349a40af1.png)

xuelongmu commented:
This looks great Ben! Merging now.
